keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/KeanuRandom.java

```diff
@@ -70,6 +71,14 @@ public double nextDouble(double min, double max) {
         return nd4jRandom.nextDouble() * (max - min) + min;
     }
 
+    public double nextDoubleNonZero() {
```

Comment:
Is this a general RNG type of function or something specific to Poisson?

Reply:
I could see this being useful in other places (in fact, I'm planning on having a look through as I suspect we potentially have a few bugs hiding where we need numbers in (0, 1) but are actually getting them in [0, 1)